lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm_2019/ashrae_90_1_prm_2019_WaterHeaterMixed.rb

```diff
@@ -0,0 +1,126 @@
+    # calculate the efficiency and skin loss coefficient.
+    # Only two cases do not use uef, which is Electricity, capacity > 40944.01
+    # and NaturalGas, capacity > 105000.01.
+    # The efficiency and skin loss coefficient of these two cases
+    # are calculated separately.
+    # The efficiency and skin loss coefficient ua is based on 90.1-2019 Table 7.8 and
+    # PNNL-23269 "Enhancements to ASHRAE Standard 90.1 Prototype Building Models" A 1.2
+    if (new_fuel == "Electricity") and (capacity_btu_per_hr > 40944.01)
+      water_heater_eff = 1
+      ua_btu_per_hr_per_f = (0.3 + 27.0/volumn_gal)/70
+    elsif (new_fuel == "NaturalGas") and (capacity_btu_per_hr > 105000.01)
+      ua_btu_per_hr_per_f = (capacity_btu_per_hr/800 + 110 * Math.sqrt(volumn_gal))*0.8/70
+      water_heater_eff = (ua_btu_per_hr_per_f * 70 + capacity_btu_per_hr * 0.8)/capacity_btu_per_hr
+    else
+      search_criteria = {}
+      search_criteria['template'] = template
+      search_criteria['fuel_type'] = new_fuel
+      if new_fuel == "Electricity"
+        search_criteria['product_class'] = "Water Heaters"
+      else
+        search_criteria['product_class'] = "Storage Water Heater"
+      end
+
+      # Todo: Use 'medium' as draw_profile for now.
+      search_criteria['draw_profile'] = "medium"
+      wh_props = model_find_object(standards_data['water_heaters'], search_criteria, capacity = capacity_btu_per_hr, date = nil, area = nil, num_floors = nil, fan_motor_bhp = nil, volume = volumn_gal)
+      unless wh_props
+        OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.WaterHeaterMixed', "For #{water_heater_mixed.name}, cannot find water heater properties, cannot apply efficiency standard.")
+        return false
+      end
+      uniform_energy_factor_base = wh_props['uniform_energy_factor_base']
+      uniform_energy_factor_volume_allowance = wh_props['uniform_energy_factor_volume_allowance']
+      uef = uniform_energy_factor_base - uniform_energy_factor_volume_allowance * volumn_gal
+      ef = water_heater_convert_uniform_energy_factor_to_energy_factor(fuel_type = new_fuel, uef = uef, capacity_btu_per_hr = capacity_btu_per_hr, volume_gal = volumn_gal)
+      eff_ua = water_heater_convert_energy_factor_to_thermal_efficiency_and_ua(new_fuel, ef, capacity_btu_per_hr)
+      water_heater_eff = eff_ua[0]
+      ua_btu_per_hr_per_f = eff_ua[1]
+    end
+
+    # Ensure that efficiency and UA were both set
+    if water_heater_eff.nil?
+      OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.WaterHeaterMixed', "For #{water_heater_mixed.name}, cannot calculate efficiency, cannot apply efficiency standard.")
+      return false
+    end
+
+    if ua_btu_per_hr_per_f.nil?
+      OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.WaterHeaterMixed', "For #{water_heater_mixed.name}, cannot calculate UA, cannot apply efficiency standard.")
+      return false
+    end
```

Comment:
The above two checks will never reached because the if-else statement covers all scenarios. It is OK to skip the check.

Reply:
To avoid duplication, use the existing water_heater_mixed_apply_efficiency function in Standards.WaterHeaterMixed.rb instead of developing a new function.